app/scripts/controllers/newOverview.js

```diff
@@ -0,0 +1,1200 @@
+    groupPipelineBuildConfigsByDeploymentConfig();
+    matchOutputImagesToImageChangeTriggers();
+  };
+
+  var groupRecentBuildsByDeploymentConfig = function() {
+    _.each(overview.deploymentConfigs, updateRecentBuildsForDeploymentConfig);
+  };
+
+  var groupBuilds = function() {
+    if(!state.builds || !overview.buildConfigs) {
+      return;
+    }
+
+    // Reset these maps.
+    overview.recentPipelinesByBuildConfig = {};
+    state.recentBuildsByBuildConfig = {};
+    state.recentPipelinesByDeploymentConfig = {};
+
+    var recentByConfig = {};
+    _.each(BuildsService.interestingBuilds(state.builds), function(build) {
+      var bcName = getBuildConfigName(build);
+      if(isJenkinsPipelineStrategy(build)) {
+        groupPipelineByDC(build);
+      } else {
+        recentByConfig[bcName] = recentByConfig[bcName] || [];
+        recentByConfig[bcName].push(build);
+      }
+    });
+
+    state.recentBuildsByBuildConfig = _.mapValues(recentByConfig, function(builds) {
+      return BuildsService.sortByBuildNumber(builds, true);
+    });
+    groupRecentBuildsByDeploymentConfig();
+  };
+
+  var updateQuotaWarnings = function() {
+    ResourceAlertsService.setGenericQuotaWarning(state.quotas,
+                                                 state.clusterQuotaData,
+                                                 $routeParams.project,
+                                                 state.alerts);
+  };
+
+  overview.clearFilter = function() {
+    LabelFilter.getLabelSelector().clearConjuncts();
+    overview.filterText = '';
+  };
+
+  $scope.$watch(function() {
+    return overview.filterText;
+  }, _.debounce(function(text, previous) {
+    if (text === previous) {
+      return;
+    }
+    state.filterKeywords = KeywordService.generateKeywords(text);
+    $scope.$evalAsync(updateFilter);
+  }, 50, { maxWait: 250 }));
+
+  $scope.$watch(function() {
+    return overview.filterBy;
+  }, function() {
+    // Clear any existing filter when switching filter types.
+    overview.clearFilter();
+    updateFilter();
+  });
+
+  LabelFilter.onActiveFiltersChanged(function() {
+    $scope.$evalAsync(updateFilter);
+  });
+
+  overview.startBuild = function(buildConfig) {
+    BuildsService
+      .startBuild(buildConfig.metadata.name, { namespace: buildConfig.metadata.namespace })
+      .then(_.noop, function(result) {
+        var buildType = isJenkinsPipelineStrategy(buildConfig) ? 'pipeline' : 'build';
+        state.alerts["start-build"] = {
+          type: "error",
+          message: "An error occurred while starting the " + buildType + ".",
+          details: getErrorDetails(result)
+        };
+      });
+  };
+
+  var watches = [];
+  ProjectsService.get($routeParams.project).then(_.spread(function(project, context) {
+    // Project must be set on `$scope` for the projects dropdown.
+    $scope.project = project;
+    state.context = context;
+
+    var updateReferencedImageStreams = function() {
```

Comment:
Ok probably something we should refactor later then, it doesn't really need the context, i think it really just needs to know the project